include/psen_scan_v2/reply_msg_from_scanner.h

```diff
@@ -41,23 +41,30 @@ class ReplyMsgFromScanner
 public:
   static ReplyMsgFromScanner fromRawData(const RawScannerData& data);
 
-public:
-  ReplyMsgFromScanner(const uint32_t op_code, const uint32_t res_code);
+private:
+  template <typename T>
+  static void read(std::istringstream& is, const T& data);
 
 public:
+  ReplyMsgFromScanner(const uint32_t op_code, const uint32_t res_code);
   ReplyMsgFromScannerType type() const;
-
-public:
   static uint32_t getStartOpCode();
   static uint32_t calcCRC(const ReplyMsgFromScanner& msg);
 
+private:
+  template <typename T>
+  static void process_bytes(boost::crc_32_type& crc_32, const T& data);
```

Comment:
You are "misleading" the reader/user of the function here because as stated in the [docu](https://en.cppreference.com/w/cpp/io/basic_istream/read) of the `std::istringstream::read()` function, it does not guarantee that the `data` argument is not changed (because the `data` argument is not defined as `const char*` argument). 
Your code only works because a C-Cast is used here which not only casts to a different pointer type but also casts away the constness of the `data` argument. If you use a `reinterpret_cast`, the compiler will complain about this fact.
My suggestion is: Remove the `const` from the `data` argument and use the `reinterpret_cast` instead of the C-cast.